design/trait-work-as-patch-of-workload.md

```diff
@@ -0,0 +1,256 @@
+# Trait Work as Patch of Workload
```

Comment:
The name of "patch trait" is a bit confusing.

A trait controller can also patch the Component CR, can patch Workload CR, instead of child resource. Are they patch traits? 

Also, when should the trait controller patch Component CR? when should patch Workload CR? When should patch child resource? What's the best practice here?

Answering these questions and clear definition of "patch trait" (ref: xref: https://github.com/crossplane/oam-kubernetes-runtime/pull/104/files#r456239518 ) is the most important part in this proposal.